### PR TITLE
chore: Update AlertmanagerConfig example

### PR DESCRIPTION
**Pull request by youngmoneee**
## Description

Update example code to enhance understanding of AlertmanagerConfig resource functionality.

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: example/user-guides/alerting/alertmanager-config-example.yaml [hunk 1]
```diff
@@ -5,13 +5,57 @@ metadata:
...
-    receiver: 'webhook'
+    receiver: webhook
+    # Alerts are normally sent via webhook,
+    # but if an alert with severity 'critical' occurs,
+    # it will be sent through Telegram.
+    routes:
+    - match:
+      severity: critical
+      receiver: telegram
+      repeatInterval: 10m
```
Comment (simonpasquier) [inline example/user-guides/alerting/alertmanager-config-example.yaml:44]: wrong indent
```suggestion
    receiver: telegram
    repeatInterval: 10m
```

File: example/user-guides/alerting/alertmanager-config-example.yaml [hunk 2]
```diff
@@ -5,13 +5,57 @@ metadata:
...
     groupInterval: 5m
     repeatInterval: 12h
-    receiver: 'webhook'
+    receiver: webhook
+    # Alerts are normally sent via webhook,
+    # but if an alert with severity 'critical' occurs,
+    # it will be sent through Telegram.
+    routes:
+    - match:
+      severity: critical
```
Comment (simonpasquier) [inline example/user-guides/alerting/alertmanager-config-example.yaml:42]:
```suggestion
    - matchers:
      - name: severity
        value: critical
        matchType: "="
```

File: example/user-guides/alerting/alertmanager-config-example.yaml [hunk 3]
```diff
@@ -5,13 +5,57 @@ metadata:
...
+      - name: severity
+        value: info
+    equal: [job]
+  - sourceMatch:
+    - name: severity
+      value: warning
+    targetMatch:
+      - name: severity
+        value: info
+    equal: [job]
```
Comment (simonpasquier) [inline example/user-guides/alerting/alertmanager-config-example.yaml:29]: to keep it as short as possible, I'd remove the second inhibition rule
```suggestion
```

File: example/user-guides/alerting/alertmanager-config-example.yaml [hunk 4]
```diff
@@ -5,13 +5,57 @@ metadata:
...
   receivers:
-  - name: 'webhook'
+  - name: webhook
     webhookConfigs:
-    - url: 'http://example.com/'
+    - url: http://example.com/
+  - name: telegram
+    telegramConfigs:
+    - chatID: -123456789
+    # Import bot-token from a pre-generated telegram-secret.
```
Comment (simonpasquier) [inline example/user-guides/alerting/alertmanager-config-example.yaml:53]:
```suggestion
    # The bot token comes from the "token" key of the "telegram" secret (in the same namespace as the AlertmanagerConfig object). 
```

File: example/user-guides/alerting/alertmanager-config-example.yaml (hunk 1 above)
Comment (youngmoneee) [reply example/user-guides/alerting/alertmanager-config-example.yaml:44]: When applying this resource, there are no issues caused by indentation.
Screenshot 2024-01-25 at 11 49 09 PM
This screenshot is actually a config applied to the Alertmanager server after the example is applied.

File: example/user-guides/alerting/alertmanager-config-example.yaml (hunk 4 above)
Comment (youngmoneee) [reply example/user-guides/alerting/alertmanager-config-example.yaml:53]: That's much more useful. Thank you!

File: example/user-guides/alerting/alertmanager-config-example.yaml (hunk 1 above)
Comment (simonpasquier) [reply example/user-guides/alerting/alertmanager-config-example.yaml:39]: my bad.

File: example/user-guides/alerting/alertmanager-config-example.yaml [hunk 5]
```diff
@@ -5,13 +5,52 @@ metadata:
...
+  # and an alert with severity 'info' occur at the same time,
+  # the alerts with severity 'info' will be ignored.
+  - sourceMatch:
+    - name: severity
+      value: critical
+    targetMatch:
+      - name: severity
+        value: warning
+      - name: severity
+        value: info
```
Comment (simonpasquier) [inline example/user-guides/alerting/alertmanager-config-example.yaml:21]: (nit) consistent indentation
```suggestion
    targetMatch:
    - name: severity
      value: warning
    - name: severity
      value: info
```

File: example/user-guides/alerting/alertmanager-config-example.yaml [hunk 6]
```diff
@@ -5,13 +5,52 @@ metadata:
   labels:
     alertmanagerConfig: example
 spec:
+  inhibitRules:
+  # critical > warning > info
+  #
+  # If an alert with severity 'warning'
+  # and an alert with severity 'info' occur at the same time,
+  # the alerts with severity 'info' will be ignored.
```
Comment (simonpasquier) [inline example/user-guides/alerting/alertmanager-config-example.yaml:13]:
```suggestion
  # If an alert with severity 'warning' occurs at the same time as an alert 
  # with severity 'critical' and both alerts share the same job label, the former will be muted.
```

File: example/user-guides/alerting/alertmanager-config-example.yaml [hunk 7]
```diff
@@ -5,13 +5,52 @@ metadata:
   labels:
     alertmanagerConfig: example
 spec:
+  inhibitRules:
+  # critical > warning > info
+  #
```
Comment (simonpasquier) [inline example/user-guides/alerting/alertmanager-config-example.yaml:10]: I'd simplify and only show critical > warning.
```suggestion
```

File: example/user-guides/alerting/alertmanager-config-example.yaml (hunk 6 above)
Comment (youngmoneee) [reply example/user-guides/alerting/alertmanager-config-example.yaml]: I modified it.
It looks much better than the first time. Thank you

File: example/user-guides/alerting/alertmanager-config-example.yaml [hunk 8]
```diff
@@ -5,13 +5,49 @@ metadata:
...
+  # If an alert with severity 'warning' occurs at the same time as an alert 
+  # with severity 'critical' and both alerts share the same job label, the former will be muted.
+  - sourceMatch:
+    - name: severity
+      value: critical
+    targetMatch:
+    - name: severity
+      value: warning
+    equal: [job]
+
```
Comment (simonpasquier) [inline example/user-guides/alerting/alertmanager-config-example.yaml:20]: (nit)
```suggestion
```

File: example/user-guides/alerting/prometheus-example-rules.yaml [hunk 9]
```diff
@@ -12,3 +12,19 @@ spec:
     rules:
     - alert: ExampleAlert
       expr: vector(1)
+  - name: node-status
+    rules:
+    - alert: DiskPressureWarning
+      expr: sum(kube_node_status_condition{condition='DiskPressure', status='true'} == 1) > 0
```
Comment (simonpasquier) [inline example/user-guides/alerting/prometheus-example-rules.yaml:18]: It assumes that Prometheus collects the kube_node_* metrics which may not be the case. I suppose that your idea is to show an example with alerts at different severities?

File: example/user-guides/alerting/prometheus-example-rules.yaml [hunk 10]
```diff
@@ -12,3 +12,19 @@ spec:
     rules:
     - alert: ExampleAlert
       expr: vector(1)
```
Comment (simonpasquier) [inline example/user-guides/alerting/prometheus-example-rules.yaml:14]: This is the simplest example that would work:
```suggestion
    rules:
    - alert: ExampleAlert
      expr: vector(1)
      labels:
        severity: critical
    - alert: ExampleAlert
      expr: vector(1)
      labels:
        severity: warning
```

Comment (youngmoneee) [reply example/user-guides/alerting/prometheus-example-rules.yaml:14]: Thank you. I will apply your feedback immediately. 👍

Comment (youngmoneee) [reply example/user-guides/alerting/prometheus-example-rules.yaml:14]: I have applied your advice and left two lines of annotation in the critical section that are used in AlertmanagerConfig.
What do you think about this change?

**Pull request closed without merging**